src/main/java/com/example/chat/groupchatbackend/controllers/MessagesController.java

```diff
@@ -27,15 +29,21 @@
     @Autowired
     private MessagesRepository messagesRepository;
     @Autowired
+    private ReadStatusRepository readStatusRepository;
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
     private UserContext userContext;
 
     @GetMapping("/messages/channel/{id}")
     public ResponseEntity getAllMessages(@PathVariable int id , @RequestParam(required = false) Long timestamp) {
         Conversation conversation = conversationsRepository.findById(id).orElseThrow(() -> new RuntimeException("conversation doesn't exist"));
         if (conversation.getConversationType().equals(ConversationType.CHANNEL)) {
             LocalDateTime date = LocalDateTime.MIN;
+            updateReadStatus(conversation, date);
```

Comment:
it aaaaaalways should be `LocalDateTime.now()`, my bad, I told it wrong before.

pleasae change `updateReadStatus` to take only one parameter (conversation) and always update with current time